src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java

```diff
@@ -506,6 +507,11 @@ public NeoForgeMod(IEventBus modEventBus, Dist dist) {
         TierSortingRegistry.init();
         if (dist == Dist.CLIENT) ClientCommandHandler.init();
         DualStackUtils.initialise();
+
+        modEventBus.addListener(CapabilityHooks::registerVanillaProviders);
+        NeoForge.EVENT_BUS.addListener(CapabilityHooks::invalidateCapsOnChunkLoad);
+        NeoForge.EVENT_BUS.addListener(CapabilityHooks::invalidateCapsOnChunkUnload);
+        NeoForge.EVENT_BUS.addListener(CapabilityHooks::cleanCapabilityListenerReferencesOnTick);
```

Comment:
Should we put these listeners on a different priority level (`HIGHEST` or `LOWEST`), in an effort to ensure these are done before/after other listeners?

Reply:
It's hard to predict what this would change in practice. I added a comment to mention that this could be re-evaluated in the future.